awx/api/metadata.py

```diff
@@ -258,6 +277,15 @@ def determine_metadata(self, request, view):
             if hasattr(serializer, 'get_types'):
                 metadata['types'] = serializer.get_types()
 
+            # Provide tail path navigation to related associate disassociate endpoints
+            from awx.api.generics import SubListCreateAttachDetachAPIView
+
+            metadata['related_associations'] = [
+                [p.strip('$/').rsplit('/')[-1], v.model._meta.verbose_name_plural.replace(' ', '_')]
+                for p, v in all_view_iterator()
+                if issubclass(v, SubListCreateAttachDetachAPIView) and v.parent_model is serializer.Meta.model
+            ]
```

Comment:
This one API change is compact, but it is a bit of a bear.

The logic is fairly high-level, as it needs to give a related parcel of a URL. In other words, out of `/api/v2/job_templates/42/notification_templates_error/`, what the CLI truly needs is "notification_templates_error", as well as the associated model. As this example shows, the model and relationship name are unrelated variables, and other relationships will sometimes have the same related model.

The challenge internal to the Django app is that URLs and views exist in separate domains. You can't easily go from a view to a URL... again, there's a mapping problem, as a view may be used for multiple URLs (at least in the mind of Django). Since we need the URL parcel and model, and only the view has the model, we have to firstly start from URLs. Those _can_ be looped over globally, and that is exactly what this does. The complexity in `all_view_iterator` is generic to Django in general. Here, we do get into some app-specific stuff. It is very high-level, but also reliable which is the key goal.